.changeset/blue-ghosts-rule.md

```diff
@@ -0,0 +1,31 @@
+---
+"@astrojs/db": patch
```

Comment:
Just to check, this is non-breaking, the old syntax is still supported for now, right?

Reply:
What will happen to people's existing indices? Will they be dropped and recreated?

Reply:
Old syntax is supported! And it _shouldn't_ drop and recreate even when refactoring to the new syntax. We handle everything with a Zod transform that matches the original object format. I'll add a unit test to confirm

Reply:
I guess what i'm asking is, currently the user can name them whatever they want. For example `foobar_index`. But now with the automated index names, is it not going to try to add the new index and not know about the previous index?

I guess you're saying that when you do push, it will see the old index and the new index and drop the old one (assuming it doesn't match) and create a new one?

Reply:
@matthewp Yep, it will drop and recreate the indexes if you _don't_ pass a `name` option. If you pass a `name` to the new indexes that matches the old name, it'll run without any migrations 👍